### ISSUE TITLE
Prepared statement parameters ignored

### ISSUE THREAD

**Issue by akamaus** [cmu-db/peloton#575]
I'm learning how to use Peloton's API and I'm having trouble using prepared queries.

I wrote the following snippet based on code I found in peloton_root/test. It runs, but `execute_prepared()` fails on second loop iteration if we leave the line with call to method `SetParameterValues()` commented. Originally I expected passing `params` argument to `TrafficCop::ExecuteStatementPlan()` would do the trick, but looks like although `params` got stored in executer's context they're not used by `InsertExecutor::DExecute()`. Is it a bug or I'm doing all this in a wrong way?

```
#include <stdlib.h>

#include "catalog/catalog.h"
#include "tcop/tcop.h"

#include "common/logger.h"
#include "common/macros.h"
#include "common/statement.h"

#include "type/types.h"
#include "type/value_factory.h"

namespace peloton {
namespace benchmark {

constexpr int NUM_INSERTS = 2;

std::vector<StatementResult> ExecuteSQLQuery(const std::string query) {
  std::vector<StatementResult> result;
  std::vector<FieldInfo> tuple_descriptor;
  std::string error_message;
  int rows_changed;

  auto& traffic_cop = tcop::TrafficCop::GetInstance();
  // execute the query using tcop
  auto status = traffic_cop.ExecuteStatement(query, result, tuple_descriptor,
                                              rows_changed, error_message);


  if (status != ResultType::SUCCESS) {
    throw std::string("error: ") + ResultTypeToString(status) + " executing " + query;
  }
  return result;
}

static void fill_database() {
  auto res = catalog::Catalog::GetInstance()->CreateDatabase(DEFAULT_DB_NAME, nullptr);
  if (res != ResultType::SUCCESS) {
    throw "cant create db";
  }

  LOG_INFO("DB created!");

  auto result = ExecuteSQLQuery("CREATE TABLE logs("
                                "a INTEGER PRIMARY KEY,"
                                "b INTEGER,"
                                "c INTEGER"
                                ");");
  LOG_INFO("table created");
}

static void execute_prepared() {
  std::string error_message;
  auto& cop = tcop::TrafficCop::GetInstance();
  auto statement = cop.PrepareStatement("insert row", "INSERT INTO logs(a,b,c) VALUES(?, 0 ,0)", error_message);

  if (!statement) {
    throw error_message;
  }
  LOG_INFO("insert prepared");

  std::vector<StatementResult> result;
  std::vector<int> result_format(statement->GetTupleDescriptor().size(), 0);

  size_t thread_id = 0;

  for (int i=0; i<NUM_INSERTS; i++) {
    std::vector<type::Value> params;
    params.push_back(type::ValueFactory::GetIntegerValue(i+1));
    // UNCOMMENT THIS:
    // statement->GetPlanTree().get()->SetParameterValues(&params);
    auto status = cop.ExecuteStatementPlan(statement->GetPlanTree().get(), params,
                                           result, result_format,
                                           thread_id);

    if (status.m_result != ResultType::SUCCESS) {
      throw std::string("error: ") + ResultTypeToString(status.m_result) + " executing insert";
    }
    LOG_INFO("insert executed %d, %s", i, statement->GetInfo().c_str());
  }
}

}
}

int main( ) {
  try {
    peloton::benchmark::fill_database();
    peloton::benchmark::execute_prepared();
    return 0;
  } catch (std::string e) {
    std::cerr << e;
    return 1;
  }
}

**Comment by apavlo** [comment]
@akamaus Let us discuss this on Friday and get back to you. We are in the process of rewriting a lot of the prepared statement handling code.

**Comment by apavlo** [comment]
I think this has been fixed in #617
